### PR TITLE
Added ngspice package

### PR DESCRIPTION
**Pull request by orsonmmz**

### CODE REVIEW

File: mingw-w64-ngspice/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+              --enable-cider \
+              --enable-openmp
+  make
+}
+
+package() {
+  cd "${srcdir}/build-${CARCH}"
+  make install DESTDIR="${pkgdir}"
+  install -D -m755 "src/.libs/libngspice-0.dll" "${pkgdir}/${MINGW_PREFIX}/bin"
+  install -D -m644 "${srcdir}/${_realname}-${pkgver}/COPYING" "${pkgdir}/usr/share/licenses/${_realname}/LICENSE"
```
Comment (Alexpux) [inline mingw-w64-ngspice/PKGBUILD]: No "usr" folder for mingw packages

File: mingw-w64-ngspice/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  install -D -m644  "${srcdir}/${_realname}-doc-${pkgver}/manual.pdf" "${pkgdir}/usr/share/doc/${_realname}/manual.pdf"
```
Comment (Alexpux) [inline mingw-w64-ngspice/PKGBUILD]: The same

File: mingw-w64-ngspice/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  ../${_realname}-${pkgver}/configure --prefix=${MINGW_PREFIX} \
+              --build=${MINGW_CHOST} \
+              --host=${MINGW_CHOST} \
+              --enable-openmp \
+              --enable-xspice \
+              --enable-cider \
+              --with-ngshared
+  make
+
+  ../${_realname}-${pkgver}/configure --prefix=${MINGW_PREFIX} \
```
Comment (Alexpux) [inline mingw-w64-ngspice/PKGBUILD]: You need build shared/static in separate folders

**Comment by orsonmmz** [comment]
Currently it is broken due to sourceforge.net failure, I will update the request when the website is up.

File: mingw-w64-ngspice/PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+#   Contributor: Kyle Keen <keenerd@gmail.com>
+#   Contributor: Abhishek Dasgupta <abhidg@gmail.com>
+#   Contributor: Jason Taylor <jftaylor21@gmail.com>
+#   Contributor: Luis Henrique <lmello.009@gmail.com>
+
+_realname=ngspice
+pkgbase=mingw-w64-${_realname}
+pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
+pkgver=26
+pkgrel=3
```
Comment (Alexpux) [inline mingw-w64-ngspice/PKGBUILD]: Reset to 1

**Comment by orsonmmz** [comment]
Thank you for the comments, everything should be fixed now. Unfortunately, the sourceforge.net problem still holds.

File: mingw-w64-ngspice/PKGBUILD [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+_realname=ngspice
+pkgbase=mingw-w64-${_realname}
+pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
+pkgver=26
+pkgrel=1
+pkgdesc='Mixed-level/Mixed-signal circuit simulator based on Spice3f5, Ciber1b1, and Xspice.'
+url='http://ngspice.sourceforge.net'
+license=('BSD')
+arch=('i686' 'x86_64')
```
Comment (Alexpux) [inline mingw-w64-ngspice/PKGBUILD]: For mingw packages always ````arch=('any')````

File: mingw-w64-ngspice/PKGBUILD [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+build() {
+  cd "${srcdir}/${_realname}-${pkgver}"
+  [[ -d "${srcdir}/build-shared-${CARCH}" ]] && rm -rf "${srcdir}/build-shared-${CARCH}"
+  mkdir -p "${srcdir}/build-shared-${CARCH}" && cd "${srcdir}/build-shared-${CARCH}"
+
+  # FS#45230, create so lib
+  # shared lib sets flags and modifies headers, needs dedicated pass
+  # adding --with-readline disables libngspice-0.dll
+  ../${_realname}-${pkgver}/configure --prefix=${MINGW_PREFIX} \
```
Comment (Alexpux) [inline mingw-w64-ngspice/PKGBUILD]: Move ````--prefix```` to next line

File: mingw-w64-ngspice/PKGBUILD [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+              --enable-xspice \
+              --enable-cider \
+              --with-ngshared
+  make
+
+  cd "${srcdir}/${_realname}-${pkgver}"
+  [[ -d "${srcdir}/build-static-${CARCH}" ]] && rm -rf "${srcdir}/build-static-${CARCH}"
+  mkdir -p "${srcdir}/build-static-${CARCH}" && cd "${srcdir}/build-static-${CARCH}"
+
+  ../${_realname}-${pkgver}/configure --prefix=${MINGW_PREFIX} \
```
Comment (Alexpux) [inline mingw-w64-ngspice/PKGBUILD]: The same

File: mingw-w64-ngspice/PKGBUILD [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+package() {
+  cd "${srcdir}/build-static-${CARCH}"
+  make install DESTDIR="${pkgdir}"
+  install -D -m644 "${srcdir}/${_realname}-${pkgver}/COPYING" "${pkgdir}${MINGW_PREFIX}/share/licenses/${_realname}/LICENSE"
+  install -D -m644  "${srcdir}/${_realname}-doc-${pkgver}/manual.pdf" "${pkgdir}${MINGW_PREFIX}/share/doc/${_realname}/manual.pdf"
+
+  cd "${srcdir}/build-shared-${CARCH}"
+  install -D -m755 "src/.libs/libngspice-0.dll" "${pkgdir}/${MINGW_PREFIX}/bin"
```
Comment (Alexpux) [inline mingw-w64-ngspice/PKGBUILD]: No need slash before MINGW_PREFIX

**Comment by Alexpux** [comment]
Will you fix my comments?

**Comment by orsonmmz** [comment]
Sure, done. BTW: is there a guideline for mingw pkgbuilds that I could follow?
I am not experiencing the sourceforge.net problem anymore on my machines, so if it persists than it might be worth further investigation.

**Comment by orsonmmz** [comment]
Thank you!

**Pull request merged**